src/cryptography/x509/extensions.py

```diff
@@ -437,8 +437,13 @@ def __init__(self, full_name, relative_name, reasons, crl_issuer):
                     "full_name must be a list of GeneralName objects"
                 )
 
-        if relative_name and not isinstance(relative_name, Name):
-            raise TypeError("relative_name must be a Name")
+        if (
+            relative_name and
+            not isinstance(relative_name, RelativeDistinguishedName)
```

Comment:
This should probably accept `Name` still, but have it be deprecated, to avoid breaking backwards compatibility.